source/collectBy.js

```diff
@@ -0,0 +1,37 @@
+import _curry2 from './internal/_curry2';
+import groupBy from './groupBy';
+import values from './values';
+
+/**
+ * Splits a list into sub-lists, based on the result of calling a String-returning function on each element,
+ * and grouping the results according to values returned.
+ *
+ * @func
+ * @memberOf R
+ * @category List
+ * @sig (a -> String) -> [a] -> {String: [a]}
+ * @param {Function} fn Function :: a -> String
+ * @param {Array} list The array to split into sub-arrays.
+ * @return {Array}
+ *    An array of arrays where each sub-array contains items for which
+ *    the String-returning function has returned the same value.
+ * @see R.groupBy
+ * @example
+ *      R.collectBy(R.prop('type'), [
+ *        {type: 'breakfast', item: '☕️'},
+ *        {type: 'lunch', item: '🌯'},
+ *        {type: 'dinner', item: '🍝'},
+ *        {type: 'breakfast', item: '🥐'},
+ *        {type: 'lunch', item: '🍕'}
+ *      ]);
```

Comment:
I think these should be in a slightly different order so that the result could not be mistaken for something equivalent to `compose(transpose, splitEvery(3))`

Reply:
Ok will do

Reply:
Added more objects and "shuffled" the array to avoid seeing patterns where there are none. What do you think?